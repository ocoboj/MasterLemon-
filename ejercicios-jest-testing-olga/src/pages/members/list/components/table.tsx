import * as React from 'react';
import { Member } from '../viewModel';
import { Header } from './header';
import { Body } from './body';

interface Props {
  members: Member[];
}

export const Table: React.StatelessComponent<Props> = (props) => (
  <table className="table table-striped">
    <Header />
    <Body members={props.members} />
  </table>
);
