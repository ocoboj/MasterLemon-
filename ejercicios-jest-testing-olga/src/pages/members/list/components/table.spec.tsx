import * as React from 'react';
import { mount } from 'enzyme';
import { Table } from './table';



describe('pages/members/list/components/table spec', () => {

    it('should render as expected when passing one item', () => {

        const prop = {
            title: 'Testing Header',

            members:[
                {
                    id: 8,
                    name: 'name test 8',
                    avatarUrl: 'url test 8'
                }
            ]
        }

        const component = mount(<Table {...prop}/>);

        expect(component).toMatchSnapshot();
    });

    it('should render as expected when passing more items', () => {
        const prop = {
            title: 'Testing Header',

            members: [
                {
                    id: 3,
                    name: 'name test 3',
                    avatarUrl: 'url test 3'
                },
                {
                    id: 7,
                    name: 'name test 7',
                    avatarUrl: 'url test 7'
                },
                {
                    id: 5,
                    name: 'name test 5',
                    avatarUrl: 'url test 5'
                },
                {
                    id: 4,
                    name: 'name test 4',
                    avatarUrl: 'url test 4'
                },
            ]
        }
    })
})