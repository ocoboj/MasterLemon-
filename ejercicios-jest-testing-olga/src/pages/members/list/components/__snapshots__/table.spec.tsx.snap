// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/members/list/components/table spec should render as expected when passing one item 1`] = `
<Component
  members={
    Array [
      Object {
        "avatarUrl": "url test 8",
        "id": 8,
        "name": "name test 8",
      },
    ]
  }
  title="Testing Header"
>
  <table
    className="table table-striped"
  >
    <Component>
      <thead>
        <tr>
          <th>
            Avatar
          </th>
          <th>
            Id
          </th>
          <th>
            Name
          </th>
        </tr>
      </thead>
    </Component>
    <Component
      members={
        Array [
          Object {
            "avatarUrl": "url test 8",
            "id": 8,
            "name": "name test 8",
          },
        ]
      }
    >
      <tbody>
        <Component
          key="8"
          member={
            Object {
              "avatarUrl": "url test 8",
              "id": 8,
              "name": "name test 8",
            }
          }
        >
          <tr>
            <td>
              <img
                src="url test 8"
                style={
                  Object {
                    "maxWidth": "10rem",
                  }
                }
              />
            </td>
            <td>
              <span>
                8
              </span>
            </td>
            <td>
              <span>
                name test 8
              </span>
            </td>
          </tr>
        </Component>
      </tbody>
    </Component>
  </table>
</Component>
`;
